source/crud/crud.md

```diff
@@ -768,7 +968,24 @@ filter, skip, and limit are added as options to the `aggregate` command.
 In the event this aggregation is run against an empty collection, an empty array will be returned with no `n` field.
 Drivers MUST interpret this result as a `0` count.
 
-##### Combining Limit and Batch Size for the Wire Protocol
+##### findOne API details
+
+The `findOne` operation is implemented using a `find` operation, but only returns the first document returned in the
+cursor. Due to the special case, `findOne` does not support the following options supported in `find`:
+
+- `batchSize`: drivers MUST NOT set a `batchSize` in the `find` command created for a `findOne` operation
+- `cursorType`: `findOne` only supports non-tailable cursors
+- `limit`: drivers MUST set `limit` to 1 in the `find` command created for a `findOne` operation
+- `noCursorTimeout`: with a `limit` of 1 and no `batchSize`, there will not be an open cursor on the server
+
+##### Setting limit and batchSize options for find commands
+
+When users specify both `limit` and `batchSize` options with the same value, the server returns all results in the
+first batch, but still leaves an open cursor that needs to be closed using the `killCursors` command. To avoid this,
+drivers MUST send a value of `limit + 1` for `batchSize` in the resulting `find` command. This eliminates the open
+cursor issue.
```

Comment:
Unlike suggested in the ticket, we can't just omit the `batchSize` option when `limit <= batchSize`. Omitting the batch size makes us fall back to the default batch size, which may be less than the limit requested by the user, defeating the expectation that all results are returned in a single batch. The only workaround for this is to set a batch size higher than limit to get the server to close the cursor immediately before returning results.